src/types/onyx/ReimbursementAccountDraft.ts

```diff
@@ -63,7 +63,10 @@ type ReimbursementAccountProps = {
     amount3?: string;
 };
 
-type ReimbursementAccountDraft = BankAccountStepProps & CompanyStepProps & RequestorStepProps & ACHContractStepProps & ReimbursementAccountProps;
+// BeneficialOwnerDraftData is saved under dynamic key which consists of prefix, beneficial owner ID and input key
```

Comment:
You can take a look at [this](https://github.com/Expensify/App/blob/aa1d87caddb6831061f3750bd78f77b42603e5e3/src/libs/actions/FormActions.ts#L23C3-L23C3) I think its solving similar type problem

Reply:
@kubabutkiewicz @MrMuzyk Maybe we can do like this?

```ts
type BeneficialOwnerDraftData = Record<`beneficialOwner_${string}`, string>;

const prefix = 'beneficialOwner_'
const id = 'something'

const data: BeneficialOwnerDraftData = {
  [`${prefix}${id}`]: 'data1',
  [`beneficialOwner_${id}`]: 'data2',
  beneficialOwner_another: 'data3',
  otherPrefix_something: 'data4' // TS error
}
```

Reply:
Can narrow it down even more then as they key consists of three parts

`    const firstNameInputID = `${BENEFICIAL_OWNER_PREFIX}_${beneficialOwnerBeingModifiedID}_${FIRST_NAME}`;
`

But I get the idea, thanks